### PR TITLE
Plugin manager changes

### PR DESCRIPTION
**Pull request by BaronGreenback**
Works alongside See https://github.com/jellyfin/jellyfin/pull/4709.

This PR basically adds the enable/disable/rollback options for the user - as well as providing the visual status of each plugin.

_Plugin status images are demo's, are totally optional, and can be changed._

image

### CODE REVIEW

File: src/controllers/dashboard/plugins/installed/index.js [hunk 1]
```diff
@@ -57,9 +77,9 @@ function getPluginCardHtml(plugin, pluginConfigurationPages) {
         html += '</div>';
     }
 
-    html += "<div class='cardText'>";
+    html += "<div class='cardText'>"
     html += configPage && configPage.DisplayName ? configPage.DisplayName : plugin.Name;
-    html += '</div>';
+    html += '<br/>Status: ' + plugin.Status +'</div>';
```
Comment (thornbill) [inline src/controllers/dashboard/plugins/installed/index.js:82]: Status should be translatable.

**Review by thornbill — CHANGES_REQUESTED** [review]
Needs merge conflict and linting fixes. Looks good otherwise!

**Comment by thornbill** [comment]
One other question... are there apiclient changes required for this?

File: src/controllers/dashboard/plugins/installed/index.js (hunk 1 above)
Comment (BaronGreenback) [reply src/controllers/dashboard/plugins/installed/index.js:82]: Could point me in the direction of how this can be done?

**Comment by BaronGreenback** [comment]
> One other question... are there apiclient changes required for this?

Nope - there are a couple of additional ones - but no breaking changes.

**Review by thornbill — APPROVED** [review]
Looks good!

Just to note this is dependent on https://github.com/jellyfin/jellyfin-apiclient-javascript/pull/104 so it will need a new version of the apiclient published and included in web for this to work.

File: src/controllers/dashboard/plugins/installed/index.js [hunk 2]
```diff
@@ -41,12 +55,18 @@ function getPluginCardHtml(plugin, pluginConfigurationPages) {
     })[0];
     const configPageUrl = configPage ? Dashboard.getPluginUrl(configPage.Name) : null;
     let html = '';
-    html += "<div data-id='" + plugin.Id + "' data-name='" + plugin.Name + "' data-removable='" + plugin.CanUninstall + "' class='card backdropCard'>";
+    html += "<div data-id='" + plugin.Id + "' data-version='" + plugin.Version + "' data-name='" + plugin.Name + "' data-removable='" + plugin.CanUninstall + "' data-status='" + plugin.Status + "' class='card backdropCard'>";
```
Comment (ferferga) [inline src/controllers/dashboard/plugins/installed/index.js:58]: Can you do this with backticks instead? https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/template_strings

File: src/controllers/dashboard/plugins/installed/index.js [hunk 3]
```diff
@@ -41,12 +55,18 @@ function getPluginCardHtml(plugin, pluginConfigurationPages) {
...
     html += '<div class="cardBox visualCardBox">';
     html += '<div class="cardScalable">';
     html += '<div class="cardPadder cardPadder-backdrop"></div>';
     html += configPageUrl ? '<a class="cardContent cardImageContainer" is="emby-linkbutton" href="' + configPageUrl + '">' : '<div class="cardContent noConfigPluginCard noHoverEffect cardImageContainer emby-button">';
-    html += '<span class="cardImageIcon material-icons folder"></span>';
+    html += '<span class="cardImageIcon';
+    if (plugin.HasImage) {
+        html += '"><img src="/Plugins/' + plugin.Id + '/' + plugin.Version + '/Image" style="width:100%;height:auto"/>';
+    } else {
+        html += ' material-icons folder">';
```
Comment (ferferga) [inline src/controllers/dashboard/plugins/installed/index.js:67]: Same here

File: src/controllers/dashboard/plugins/installed/index.js [hunk 4]
```diff
@@ -59,7 +79,7 @@ function getPluginCardHtml(plugin, pluginConfigurationPages) {
 
     html += "<div class='cardText'>";
     html += configPage && configPage.DisplayName ? configPage.DisplayName : plugin.Name;
-    html += '</div>';
+    html += '<br/>' + globalize.translate('LabelStatus') + ' ' + plugin.Status + '</div>';
```
Comment (ferferga) [inline src/controllers/dashboard/plugins/installed/index.js:82]: Same here

**Review by ferferga — CHANGES_REQUESTED** [review]
Some nitpicky stuff

File: src/controllers/dashboard/plugins/installed/index.js (hunk 2 above)
Comment (BaronGreenback) [reply src/controllers/dashboard/plugins/installed/index.js:58]: Been a long time absent from js. You mean like
```
html += "<div data-id='${plugin.Id}' data-version='${plugin.Version}' data-name='${plugin.Name}' data-removable='${plugin.CanUninstall}' data-status='${plugin.Status}' class='card backdropCard'>";
```

Comment (ferferga) [reply src/controllers/dashboard/plugins/installed/index.js:58]: @BaronGreenback Exactly like this:
```suggestion
    html += `<div data-id='${plugin.Id}' data-version='${plugin.Version}' data-name='${plugin.Name}' data-removable='${plugin.CanUninstall}' data-status='${plugin.Status}' class='card backdropCard'>`;
```

File: src/controllers/dashboard/plugins/installed/index.js (hunk 4 above)
Comment (BaronGreenback) [reply src/controllers/dashboard/plugins/installed/index.js]: didn't know if ${globalize.transtion(...) would work so started from the trailing space.

File: src/controllers/dashboard/plugins/installed/index.js [hunk 5]
```diff
@@ -59,7 +79,7 @@ function getPluginCardHtml(plugin, pluginConfigurationPages) {
 
     html += "<div class='cardText'>";
     html += configPage && configPage.DisplayName ? configPage.DisplayName : plugin.Name;
-    html += '</div>';
+    html += '<br/>' + globalize.translate('LabelStatus') + ` ${plugin.Status}</div>`;
```
Comment (ferferga) [inline src/controllers/dashboard/plugins/installed/index.js:82]:
```suggestion
    html += `<br/>${globalize.translate('LabelStatus')}${plugin.Status}</div>`;
```

Not sure what are you trying to do here is correct though

**Review by ferferga — CHANGES_REQUESTED** [review]

File: src/controllers/dashboard/plugins/installed/index.js (hunk 5 above)
Comment (dmitrylyzo) [reply src/controllers/dashboard/plugins/installed/index.js:82]: @ferferga you missed the space between `LabelStatus` and `Status`

Comment (ferferga) [reply src/controllers/dashboard/plugins/installed/index.js:82]: @dmitrylyzo Oh I see

Edited

**Review by ferferga — APPROVED** [review]

File: src/strings/en-gb.json [hunk 6]
```diff
@@ -1457,6 +1457,8 @@
     "LabelEnableIP4Help": "Enables IPv4 functionality.",
     "LabelEnableIP4": "Enable IPv4:",
     "HeaderNetworking": "IP Protocols",
+    "EnablePlugin": "Enable Plugin",
+    "DisablePlugin": "Disable Plugin",
```
Comment (joshuaboniface) [inline src/strings/en-gb.json:1461]: Please note for future @BaronGreenback that `en-us` is the translation base, not `en-gb`.

Comment (joshuaboniface) [reply src/strings/en-gb.json:1461]: I corrected this manually in the cherry-pick by including the translations in both `en-us` and `en-gb`. It needs to still be corrected in `master` in a separate PR but I'm not waiting around for that.

**Pull request merged**